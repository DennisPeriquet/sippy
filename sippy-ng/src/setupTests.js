/** @jest-environment setup-polly-jest/jest-environment-node */

import '@testing-library/jest-dom'
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import fetch from 'node-fetch'
import path from 'path'
import { setupPolly } from 'setup-polly-jest'
import toDiffableHtml from 'diffable-html'

// https://github.com/mui-org/material-ui/issues/21293
export const withoutMuiID = (wrapper) => toDiffableHtml(wrapper.html()
  .replace(/id="mui-[0-9]*"/g, '')
  .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, ''))

Enzyme.configure({ adapter: new Adapter() })

// To stub out fetch(), we need to use node-fetch and this line:
global.fetch = fetch

// See: https://github.com/vuejs/vue-test-utils/issues/974
global.requestAnimationFrame = cb => cb()

// When injecting multiple filters, material table complains.
const originalError = console.error.bind(console.error)

global.console.error = (log) => {
  if (!log.toString().includes('Warning: Failed')) {
    originalError(log.toString())
  }
}

// Set API URL for Sippy
process.env.REACT_APP_API_URL = 'http://localhost:8080'

// Default PollyJS mode is replay, set to record to record
// new API calls.
if (!process.env.POLLY_MODE) {
  process.env.POLLY_MODE = 'replay'
}

export const setupDefaultPolly = () => {
  const context = setupPolly({
    mode: process.env.POLLY_MODE,
    adapters: [require('@pollyjs/adapter-node-http')],
    persister: require('@pollyjs/persister-fs'),
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(__dirname, '../__recordings__')
      }
    }
  })
  return context
}

export const expectLoadingPage = (wrapper) => expect(wrapper.find('p').contains('Loading...'))

jest.mock('react-chartjs-2', () => ({
  Bar: () => null,
  Chart: () => null,
  ChartComponent: () => null,
  Doughnut: () => null,
  Line: () => null,
  PolarArea: () => null
}))
