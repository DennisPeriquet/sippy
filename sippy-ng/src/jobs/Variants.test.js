/** @jest-environment setup-polly-jest/jest-environment-node */

import 'jsdom-global/register'
import React from 'react'
import { act, waitFor } from '@testing-library/react'
import { mount } from 'enzyme'
import { expectLoadingPage, setupDefaultPolly, withoutMuiID } from '../setupTests'
import Variants from './Variants'

jest.useRealTimers()

describe('Variants', () => {
  setupDefaultPolly()

  it('should render correctly', async () => {
    const fetchSpy = jest.spyOn(global, 'fetch')

    let wrapper
    await act(async () => {
      wrapper = mount(<Variants release="4.8"/>)
    })

    expectLoadingPage(wrapper).toBeTruthy()
    await waitFor(() => {
      wrapper.update()
      expectLoadingPage(wrapper).toBeFalsy()
    })

    expect(wrapper.text()).toContain('gcp')
    expect(withoutMuiID(wrapper)).toMatchSnapshot()
    expect(fetchSpy).toHaveBeenCalledTimes(1)
  })
})
