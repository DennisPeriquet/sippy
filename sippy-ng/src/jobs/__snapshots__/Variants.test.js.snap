// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variants should render correctly 1`] = `
<div class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded">
  <table class="MuiTable-root"
         aria-label="collapsible table"
  >
    <thead class="MuiTableHead-root">
      <tr class="MuiTableRow-root MuiTableRow-head">
        <th class="MuiTableCell-root MuiTableCell-head"
            scope="col"
        >
        </th>
        <th class="MuiTableCell-root MuiTableCell-head"
            scope="col"
        >
          Variant
        </th>
        <th class="MuiTableCell-root MuiTableCell-head"
            scope="col"
        >
          Current Period
        </th>
        <th class="MuiTableCell-root MuiTableCell-head"
            scope="col"
        >
        </th>
        <th class="MuiTableCell-root MuiTableCell-head"
            scope="col"
        >
          Previous Period
        </th>
      </tr>
    </thead>
    <tbody class="MuiTableBody-root">
      <tr class="MuiTableRow-root top: 0px;"
          colspan="6"
      >
      </tr>
    </tbody>
  </table>
</div>
`;
